### PR TITLE
MINOR: Fixed log in Topology Builder.

### PR DESCRIPTION
**Pull request by kamalcph**
- addressed some warnings shown by Intellij

https://github.com/JetBrains/intellij-community/blob/master/plugins/InspectionGadgets/src/inspectionDescriptions/ToArrayCallWithZeroLengthArrayArgument.html

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

**Comment by kamalcph** [comment]
@mjsax 
Could you please review this minor PR ?

File: streams/src/main/java/org/apache/kafka/streams/processor/internals/InternalTopologyBuilder.java [hunk 1]
```diff
@@ -1716,9 +1716,9 @@ public String toString() {
             final StringBuilder sb = new StringBuilder();
             sb.append("Topologies:\n ");
             final TopologyDescription.Subtopology[] sortedSubtopologies =
-                subtopologies.descendingSet().toArray(new TopologyDescription.Subtopology[subtopologies.size()]);
+                subtopologies.descendingSet().toArray(new Subtopology[0]);
```
Comment (satishd) [inline streams/src/main/java/org/apache/kafka/streams/processor/internals/InternalTopologyBuilder.java]: Explanation for the change:
https://github.com/JetBrains/intellij-community/blob/master/plugins/InspectionGadgets/src/inspectionDescriptions/ToArrayCallWithZeroLengthArrayArgument.html

File: streams/src/main/java/org/apache/kafka/streams/processor/internals/InternalTopologyBuilder.java [hunk 2]
```diff
@@ -1819,7 +1819,7 @@ public String toString() {
 
     public void updateSubscribedTopics(final Set<String> topics, final String logPrefix) {
         final SubscriptionUpdates subscriptionUpdates = new SubscriptionUpdates();
-        log.debug("{}found {} topics possibly matching regex", topics, logPrefix);
+        log.debug("{}found {} topics possibly matching regex", logPrefix, topics);
```
Comment (viktorsomogyi) [inline streams/src/main/java/org/apache/kafka/streams/processor/internals/InternalTopologyBuilder.java:1822]: nit: do we need a space here?

Comment (kamalcph) [reply streams/src/main/java/org/apache/kafka/streams/processor/internals/InternalTopologyBuilder.java:1822]: `logPrefix` already contains trailing whitespace.

Comment (viktorsomogyi) [reply streams/src/main/java/org/apache/kafka/streams/processor/internals/InternalTopologyBuilder.java:1822]: I see, thanks.

**Review by viktorsomogyi — APPROVED** [review]
LGTM

**Review by satishd — APPROVED** [review]
+1 LGTM

**Review by mjsax — APPROVED** [review]
LGTM.

Thanks for the fix! Can you prepare a second PR against 2.0 branch that only contains the actual fix? I cannot cherry-pick this PR to older branches because it contains Java8 cleanup that will break Java7 branches. Thx.

**Comment by kamalcph** [comment]
@viktorsomogyi @satishd @mjsax 
Thanks for reviewing the PR. Opened a new PR #5482 against 2.0 branch.

**Comment by mjsax** [comment]
Merged to `trunk` and cherry-picked to `2.0` (2.0 dropped Java8 already :))

Thanks a lot for the PR!

**Pull request merged**